### PR TITLE
Initialise DNNClassifier with normalizer_fn 

### PR DESCRIPTION
**Pull request by Montmorency**
For Contrib added option to pass normalizer_fn to DNNClassifier. This should automatically attach the normaliser to every layer in the deep neural network. Something like:
     m = tf.contrib.learn.DNNClassifier(model_dir         = model_dir,
                                                                feature_columns   = deep_columns,
                                                                dropout           = DROPOUT,
                                                                normalizer_fn     = tf.contrib.layers.batch_norm,
                                                                normalizer_params = normalizer_params...)
  
Modified dnn.py

### CODE REVIEW

**Comment by Montmorency** [comment]
Hi All,
Just thought this might be a useful feature to have in the contrib/learn functionality. It looks like the batch_norm method in /layers can be attached straight forwardly to the layers in the dnn, along with all the batch_parameters that need to be passed along. The only pitfalls I can see are there need to be some clear warning about argument scoping, and making sure the is_training variable is set properly depending on whether the model is being fit or evaluated.

File: tensorflow/contrib/learn/python/learn/estimators/dnn.py [hunk 1]
```diff
@@ -24,6 +24,8 @@
 from tensorflow.contrib.framework import deprecated_arg_values
 from tensorflow.contrib.framework.python.ops import variables as contrib_variables
 from tensorflow.contrib.layers.python.layers import optimizers
+#Importing batchnorm to be added to the DNN.
```
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/estimators/dnn.py]: please don't add comments before imports. also we try to keep them sorted.

File: tensorflow/contrib/learn/python/learn/estimators/dnn.py [hunk 2]
```diff
@@ -124,17 +129,23 @@ def _dnn_model_fn(features, labels, mode, params):
...
         parent_scope + "/hiddenlayer_%d" % layer_id,
         values=[net],
         partitioner=hidden_layer_partitioner) as scope:
+
       net = layers.fully_connected(
           net,
           num_hidden_units,
           activation_fn=activation_fn,
-          variables_collections=[parent_scope],
+#HL think we can add here the batch_norm.
```
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/estimators/dnn.py]: I don't think this comment is needed (and in general one-line comments should have the same indentation as other code). 
Also please format the arguments the same way as others (e.g. no extra spaces).

File: tensorflow/contrib/learn/python/learn/estimators/dnn.py [hunk 3]
```diff
@@ -102,7 +105,9 @@ def _dnn_model_fn(features, labels, mode, params):
   activation_fn = params.get("activation_fn")
   dropout = params.get("dropout")
   gradient_clip_norm = params.get("gradient_clip_norm")
-  num_ps_replicas = params.get("num_ps_replicas", 0)
+  num_ps_replicas   = params.get("num_ps_replicas", 0)
```
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/estimators/dnn.py]: please no extra spaces.

**Review by ilblackdragon — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Can you also please add a few tests showing usage and testing parameters are passed correctly.

Thanks!

**Comment by drpngx** [comment]
@Montmorency sorry this fell through the cracks, could you pull rebase and push again?

**Comment by drpngx** [comment]
Closing for now. Feel free to re-open once you have made changes to the code.
